middleware/realip.go

```diff
@@ -9,12 +9,13 @@ import (
 	"strings"
 )
 
+var cfConnectionIP = http.CanonicalHeaderKey("CF-Connecting-IP")
```

Comment:
I found myself reimplementing realip middleware myself in the past just to support `CF-Connecting-IP`, which is available on Cloudflare free plan (note: `True-Client-IP` is Enterprise only).

![Screen Shot 2024-04-09 at 12 48 46 PM](https://github.com/go-chi/chi/assets/139342/ec476a6e-9d66-49b4-8a45-ee61c24553ba)

So I think this makes sense.

Reply:
However, I'm thinking that perhaps we should make this configurable rather than hardcode more provider-specific headers into chi.

I think exposing a list of headers and iterating over them in a loop would probably make sense.

```go
var RealIPHeaders = []string{"True-Client-IP", "X-Forwarded-For", "X-Real-IP"}

func realIP(r *http.Request) string {
	for _, header := RealIPHeaders {
		// check if not empty
		// remove trailing `; *` string
		// check if valid IP
		return ip
	}
	return ""
}
```

It feels like easy change to me.

@pkieltyka thoughts?